app/src/main/java/com/alexis/myanimecompanion/data/remote/MyAnimeListAPI.kt

```diff
@@ -0,0 +1,66 @@
+package com.alexis.myanimecompanion.data.remote
+
+import com.alexis.myanimecompanion.QueryFieldsBuilder
+import com.alexis.myanimecompanion.data.remote.models.*
+import retrofit2.http.*
+
+interface MyAnimeListAPI {
+    @GET("anime")
+    suspend fun search(
+        @Query("q") query: String? = null,
+        @Query("limit") limit: Int? = null,
+        @Query("offset") offset: Int? = null,
+        @Query("fields") fields: String? = null,
+    ): SearchResult
+
+    @GET("anime/{anime_id}")
+    suspend fun getAnimeDetails(
+        @Path("anime_id") animeId: Int,
+        @Query("fields") fields: String? = QueryFieldsBuilder.fieldsForAnimeDetails().done(),
```

Comment:
Is it really required to pass all fields on this API call? Isn't there an easier way to use this with default values?

Reply:
I couldn't find any alternative. From my tests, the API only returned basic fields unless additional fields were explicitly requested.

Reply:
Understood